Content.Server/GameTicking/Rules/TraitorRuleSystem.cs

```diff
@@ -171,9 +172,23 @@ private void OnPlayersSpawned(RulePlayerJobsAssignedEvent ev)
 
     public List<IPlayerSession> FindPotentialTraitors(in Dictionary<IPlayerSession, HumanoidCharacterProfile> candidates)
     {
-        var list = new List<IPlayerSession>(candidates.Keys).Where(x =>
-            x.Data.ContentData()?.Mind?.AllRoles.All(role => role is not Job { CanBeAntag: false }) ?? false
-        ).ToList();
+        var list = new List<IPlayerSession>();
+        var pendingQuery = GetEntityQuery<PendingClockInComponent>();
+
+        foreach (var player in candidates.Keys)
+        {
+            // Role prevents antag.
+            if (!(player.Data.ContentData()?.Mind?.AllRoles.All(role => role is not Job { CanBeAntag: false }) ?? false))
```

Comment:
Someone vibecheck this for me coz I hate reading it and nfi if I did it properly.